src/gas-price/gas-price.test.ts

```diff
@@ -52,17 +54,11 @@ describe('gas price', () => {
     beforeEach(() => {
       initializeGasStore(chainId, providerName);
       // Reset the gasPriceStore
-      const state = getState();
-      setState({
-        ...state,
-        gasPriceStore: {
-          ...state.gasPriceStore,
-          [chainId]: {
-            ...state.gasPriceStore[chainId],
-            [providerName]: { gasPrices: [], lastOnChainDataFeedValues: {} },
-          },
-        },
-      });
+      setState(
+        produce(getState(), (draft) => {
```

Comment:
I think this should be an internal-only detail of `setState`. Maybe call it `updateState` and accept a callback.